LiteCore/Support/WeakHolder.hh

```diff
@@ -12,29 +12,26 @@
 
 #pragma once
 #include "fleece/RefCounted.hh"
-#include <shared_mutex>
 
 namespace litecore {
 
 /** WeakHolder<T>: holds a pointer to T weakly. Unlike general weak reference, one cannot get a strong holder from it.
-    Instead, we can call the methods of class T via invoke, which returns true if the call goes through as the underlying
-    pointer is good. */
+    Instead, we can call the methods of class T via invoke, which returns true if pointer is strongly referenced by some
+    object other than *this.
+    Pre-conditions: T must be dynamically castable to RefCounted.
+    Note: WeakHolder holds a strong reference but provides no API to release it. Threfore, leaking of
+    WeakHolder also leaks the object being held. It also implies that you cannot rely on the destructor of the held object
+    to auto-release the WeakHolder.
+ */
 template <typename T>
 class WeakHolder : public RefCounted {
 public:
     WeakHolder(T* pointer)
     : _pointer(pointer)
     {
         DebugAssert(_pointer != nullptr);
-    }
-
-    // Only the original owner may rescind the pointer.
-    // After rescind(), the held pointer becomes nullptr and invoke() returns false.
-    void rescind(T* owner) {
-        if (owner == _pointer) {
-            std::lock_guard<std::shared_mutex> lock(_mutex);
-            _pointer = nullptr;
-        }
+        _holder = dynamic_cast<RefCounted*>(_pointer);
+        Assert(_holder);
```

Comment:
I feel like this is something that can be asserted at compile time somehow

`static_assert(std::is_assignable<T*, RefCounted*>::value)`

Reply:
Nevermind...the dual inheritance makes this painful....